pkg/build/api/v1beta3/conversion.go

```diff
@@ -179,10 +185,12 @@ func init() {
 			return nil
 		},
 		func(in *BuildConfigSpec, out *newer.BuildParameters, s conversion.Scope) error {
-			err := s.DefaultConvert(&in.Strategy, &out.Strategy, conversion.IgnoreMissingFields)
-			if err != nil {
+			if err := s.Convert(&in.Strategy, &out.Strategy, 0); err != nil {
 				return err
 			}
+			if in.Strategy.Type == SourceBuildStrategyType {
```

Comment:
> On May 15, 2015, at 3:00 PM, Ben Parees <notifications@github.com> wrote:
> 
> In pkg/build/api/v1beta3/conversion.go:
> 
> >  				return err
> >  			}
> > +			if in.Strategy.Type == SourceBuildStrategyType {
> Since both of these constants are "Source" i'm not clear what purpose this services...we're not doing it for any of the other types.
> 
The internal type is still STI.  The printer and describer uses the internal type.
> would make sense if you weren't changing the internal api.
> 
I'm not.  I'm changing the go struct names, not the json names.
> —
> Reply to this email directly or view it on GitHub.
>